### PR TITLE
[video detection] Added range FP32 FWD oneDNN kernel

### PR DESCRIPTION
**Pull request by jakpiase**
### PR types
New features

### PR changes
OPs

### Describe
Added range FP32 FWD oneDNN kernel for video detection models

### CODE REVIEW

**Comment by jakpiase** [comment]
@piotrekobiIntel please review

File: paddle/fluid/operators/mkldnn/range_mkldnn_op.cc [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    return this->AcquireMemoryFromPrimitive(this->fwd_pd_->src_desc(), ptr);
+  }
+
+  std::shared_ptr<dnnl::memory> AcquireDstMemory(framework::Tensor *out) {
+    T *ptr = out->data<T>();
+    return this->AcquireMemoryFromPrimitive(this->fwd_pd_->dst_desc(),
+                                            ptr + RANGE_SHIFT);
+  }
+
+  static constexpr int RANGE_SHIFT = 64 / sizeof(T);  // AVX512 stores 64 bytes
```
Comment (lidanqing-vv) [inline paddle/fluid/operators/mkldnn/range_mkldnn_op.cc:52]: - RANGE_SHIFT is required by oneDNN?
- What about avx2 machines, maybe you need some condition for this kernel ?

File: paddle/fluid/operators/range_op.cc [hunk 2]
```diff
@@ -67,6 +67,21 @@ class RangeOp : public framework::OperatorWithKernel {
     ctx->SetOutputDim("Out", {-1});
   }
 
+  framework::OpKernelType GetExpectedKernelType(
+      const framework::ExecutionContext &ctx) const override {
+    auto input_data_type =
+        framework::OperatorWithKernel::IndicateVarDataType(ctx, "Start");
+
+#ifdef PADDLE_WITH_MKLDNN
+    if (this->CanMKLDNNBeUsed(ctx, input_data_type)) {
```
Comment (lidanqing-vv) [inline paddle/fluid/operators/range_op.cc:76]: Do you need to check AVX512 pre-condition here ? Could you actually paste relevant doc or manual for this kernel link here ? Thanks !

**Comment by lidanqing-vv** [comment]
Hi, @jakpiase Not review for this PR, just some comments. Last time I asked you how do we know in the python test, matmul_v2 kernel is actually used while you call paddle.matmul, you said that you refered to official document. 
But today I just found that we can find mapping relations, in nn.py file
```
@templatedoc(op_type="nearest_interp")
def resize_nearest(input,
                   out_shape=None,
                   scale=None,
                   name=None,
                   actual_shape=None,
                   align_corners=True,
                   data_format='NCHW'):
```
However, for matmul and matmul_v2 it is
```
@deprecated(since="2.0.0", update_to="paddle.matmul")
```
Do you know any good way to see python and c++ kernel mapping?

File: paddle/fluid/operators/mkldnn/range_mkldnn_op.cc [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+    this->AcquireForwardPrimitiveDescriptor(dnnl::prop_kind::forward_training,
+                                            dnnl::algorithm::eltwise_linear, md,
+                                            1.0f, RANGE_SHIFT * step);
+  }
+
+  std::shared_ptr<dnnl::memory> AcquireSrcMemory(framework::Tensor *out) {
+    T *ptr = out->data<T>();
+    return this->AcquireMemoryFromPrimitive(this->fwd_pd_->src_desc(), ptr);
+  }
```
Comment (ghost) [inline paddle/fluid/operators/mkldnn/range_mkldnn_op.cc:44]: Is this overload required? The only difference between it and the version in mkldnn_reuse.h:
`std::shared_ptr<mkldnn::memory> AcquireSrcMemory(const framework::Tensor* input) {
    const T* input_data = input->data<T>();
    return this->AcquireMemoryFromPrimitive(fwd_pd_->src_desc(),
                                            to_void_cast<T>(input_data));
  }`
is the lack of the to_void_cast. Is removing it necessary?

File: paddle/fluid/operators/mkldnn/range_mkldnn_op.cc [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+    Tensor *out = ctx.Output<Tensor>("Out");
+    out->Resize(framework::make_ddim({size}));
+    out->mutable_data<T>(ctx.GetPlace());
+
+    T *data = out->data<T>();
+    data[0] = start;
+
+    if (size <= RangeMKLDNNHandler<T>::RANGE_SHIFT) {
+      for (int i = 1; i < size; ++i) {
+        data[i] = data[i - 1] + step;
```
Comment (ghost) [inline paddle/fluid/operators/mkldnn/range_mkldnn_op.cc:78]: Seems a bit simpler to me:
```suggestion
      for (int i = 0; i < size - 1; ++i) {
        data[i] = data[i] + step;
```

File: paddle/fluid/operators/mkldnn/range_mkldnn_op.cc [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+      }
+    } else {
+      for (int i = 1; i < RangeMKLDNNHandler<T>::RANGE_SHIFT; ++i) {
+        data[i] = data[i - 1] + step;
```
Comment (ghost) [inline paddle/fluid/operators/mkldnn/range_mkldnn_op.cc:82]:
```suggestion
      for (int i = 0; i < RangeMKLDNNHandler<T>::RANGE_SHIFT - 1; ++i) {
        data[i] = data[i] + step;
```

File: paddle/fluid/operators/mkldnn/range_mkldnn_op.cc [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+
+using framework::Tensor;
+
+template <typename T>
+class RangeMKLDNNHandler
+    : public platform::MKLDNNHandlerNoCachingT<T, dnnl::eltwise_forward,
+                                               dnnl::eltwise_backward> {
+ public:
+  RangeMKLDNNHandler(const T step, const int tensor_size,
+                     const dnnl::engine engine, platform::Place cpu_place)
```
Comment (ghost) [inline paddle/fluid/operators/mkldnn/range_mkldnn_op.cc:28]: I know this is extremely minor and most mkldnn_reuse.h functions don't do it anyway, but I think non-primitive type arguments should always be passed by reference if possible
```suggestion
                     const dnnl::engine& engine, platform::Place& cpu_place)
```

**Comment by lidanqing-vv** [comment]
Sorry I need to close temporarily

**Comment by jakpiase** [comment]
Closing due to low priority and the fact that video-detection models are not available

**Pull request closed without merging**